tasklist/els-schema/src/main/java/io/camunda/tasklist/util/IndexSchemaValidatorUtil.java

```diff
@@ -97,20 +97,15 @@ public void validateDifferenceAndCollectNewFields(
     }
 
     // Validate if the difference is dynamic
-    if (difference.getEntriesDiffering() != null) {
-      for (final PropertyDifference propertyDifference : difference.getEntriesDiffering()) {
-        final Object typeDefinition = propertyDifference.getLeftValue().getTypeDefinition();
-        if (propertyDifference.getLeftValue().getTypeDefinition() instanceof Map) {
-          final Map<String, Object> typeDefMap = (Map<String, Object>) typeDefinition;
-          final Object dynamicValue = typeDefMap.getOrDefault("dynamic", false);
-          if (dynamicValue.equals(true)) {
-            LOGGER.debug(
-                String.format(
-                    "Difference is on dynamic field - continue initialization: %s.",
-                    indexDescriptor.getIndexName()));
-            return;
-          }
-        }
+    if (difference.getEntriesDiffering() != null && difference.getEntriesDiffering().size() > 0) {
+      if (difference.getEntriesDiffering().stream().noneMatch(this::nonDynamicPropertyDifference)) {
```

Comment:
🔧 Use AND instead of nesting if statements
```suggestion
    if (difference.getEntriesDiffering() != null && difference.getEntriesDiffering().size() > 0
      && difference.getEntriesDiffering().stream().noneMatch(this::nonDynamicPropertyDifference)) {
```